### PR TITLE
chore: avoid re-estimate `gasPrice` at `estimateTxDependencies`

### PR DESCRIPTION
**Pull request by Torres-ssf**
# Summary

This PR sets the `gasPrice` parameter to `0` when dry-running a transaction in `estimateTxDependencies`. This eliminates the need to re-estimate the `gasPrice` during subsequent dry-run attempts for the same method.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

**Review by arboleya — DISMISSED** [review]

File: .changeset/mean-sheep-occur.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@fuel-ts/account": minor
```
Comment (Torres-ssf) [inline .changeset/mean-sheep-occur.md:2]:
```suggestion
"@fuel-ts/account": patch
```

**Review by arboleya — APPROVED** [review]
Good catch!

Do you know if the extra request was happening only in subsequent dry-run calls?

Just trying to understand where the performance improvement should be perceived.

**Review by petertonysmith94 — APPROVED** [review]
Might be worthwhile adding a benchmark around the `sendTransaction` function where `estimateTxDependencies` is true :)

**Comment by Torres-ssf** [comment]
> Good catch!
> 
> Do you know if the extra request was happening only in subsequent dry-run calls?
> 
> Just trying to understand where the performance improvement should be perceived.

@arboleya I've also avoided calling the `estimateGasPrice` query when re-calculating the TX `maxFee` at the `Account.fund`method by informing the already estimated value [here](https://github.com/FuelLabs/fuels-ts/pull/3461/commits/4c564032ee548fe996f42e7b528a468239b5ce3f)

**Comment by Torres-ssf** [comment]
> Might be worthwhile adding a benchmark around the `sendTransaction` function where `estimateTxDependencies` is true :)

@petertonysmith94 For local node or test-net?

**Comment by petertonysmith94** [comment]
@Torres-ssf was thinking a benchmark within our [`@internal/benchmarks`](https://github.com/FuelLabs/fuels-ts/blob/master/internal/benchmarks) package - can't see anywhere using this flow.

**Pull request merged**